front/app/containers/Admin/settings/customize/Header/index.tsx

```diff
@@ -46,32 +55,65 @@ const LabelTooltip = styled.div`
   margin-right: 20px;
 `;
 
+const BgHeaderPreviewSelect = styled(Select)`
+  margin-bottom: 20px;
+`;
+
 interface Props {
   header_bg: UploadFile[] | null;
   headerError: string | null;
   titleError: Multiloc;
   subtitleError: Multiloc;
   latestAppConfigStyleSettings?: IAppConfigurationStyle;
-  latestAppConfigCoreSettings?: IAppConfigurationSettingsCore;
+  latestAppConfigSettings:
+    | IAppConfigurationSettings
+    | Partial<IAppConfigurationSettings>;
   setParentState: (state: any) => void;
   getSetting: (settingName: string) => any;
+  handleSettingOnChange: (
+    settingName: TAppConfigurationSetting
+  ) => (settingKey: string, settingValue: any) => void;
 }
 
 const TITLE_MAX_CHAR_COUNT = 45;
 const SUBTITLE_MAX_CHAR_COUNT = 90;
 
+export type PreviewDevice = 'mobile' | 'tablet' | 'desktop';
 const Header = ({
   header_bg,
   headerError,
   titleError,
   subtitleError,
   latestAppConfigStyleSettings,
-  latestAppConfigCoreSettings,
+  latestAppConfigSettings,
   setParentState,
   getSetting,
+  handleSettingOnChange,
   intl: { formatMessage },
 }: Props & InjectedIntlProps) => {
   const theme: any = useTheme();
+  const [previewDevice, setPreviewDevice] = useState<PreviewDevice>('desktop');
+
+  const handleOverlayColorOnChange = (color: string) => {
+    handleAppConfigurationStyleChange('signedOutHeaderOverlayColor')(color);
+  };
+
+  const handleOverlayOpacityOnChange = (opacity: number) => {
+    handleAppConfigurationStyleChange('signedOutHeaderOverlayOpacity')(opacity);
+  };
+
+  const debounceHandleOverlayOpacityOnChange = debounce(
+    handleOverlayOpacityOnChange,
+    15
+  );
```

Comment:
The overlay opacity slider with preview is very CPU-heavy. This debounce improves things a bit already, but still looking for better solutions.